source/validate.cpp

```diff
@@ -518,6 +436,109 @@ void DebugInstructionPass(ValidationState_t& _,
   }
 }
 
+// Allows for piping of multiple functions which return spv_result_t
+spv_result_t operator||(spv_result_t lhs, spv_result_t rhs) {
+  if (lhs) return lhs;
+  if (rhs) return rhs;
```

Comment:
Alternately ,replace the last two lines withe
```
return rhs;
```
Coder's choice.   Your code has a nicely suggestive parallel structure.